lightning/src/ln/channelmanager.rs

```diff
@@ -1265,21 +1279,30 @@ impl<'a> PersistenceNotifierGuard<'a, fn() -> NotifyOption> { // We don't care w
 
 	/// Note that if any [`ChannelMonitorUpdate`]s are possibly generated,
 	/// [`ChannelManager::process_background_events`] MUST be called first.
-	fn optionally_notify<F: Fn() -> NotifyOption>(lock: &'a RwLock<()>, notifier: &'a Notifier, persist_check: F) -> PersistenceNotifierGuard<'a, F> {
+	fn optionally_notify<F: FnMut() -> NotifyOption>(
+		lock: &'a RwLock<()>, notifier: &'a Notifier, needs_persist: &'a AtomicBool, persist_check: F
```

Comment:
May be cleaner at the call sites to pass the `ChannelManager` here as is done in `notify_on_drop`.